### PR TITLE
infra: migrate to typescript project references

### PR DESCRIPTION
**Pull request by AviVahl**

### CODE REVIEW

File: packages/experimental-loader/src/create-runtime-target-code.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { StylableExports } from '@stylable/core';
 
 export function createRuntimeTargetCode(namespace: string, mapping: StylableExports) {
     return `
-  var rt = require("@stylable/runtime/cjs/css-runtime-stylesheet.js");
+  var rt = require("@stylable/runtime");
```
Comment (barak007) [inline packages/experimental-loader/src/create-runtime-target-code.ts:5]: use "@stylable/runtime/dist/css-runtime-stylesheet.js" we don't want the renderer to be included

Comment (AviVahl) [reply packages/experimental-loader/src/create-runtime-target-code.ts]: Done.

File: packages/e2e-test-kit/src/browser-functions.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export function filterAssetResponses(responses: Response[], assetNames: string[]
 function getStyleElementsMetadata({
     includeRuntimeId,
     includeCSSContent,
-}: void | { includeCSSContent?: boolean; includeRuntimeId?: boolean } = {}) {
+}: any | { includeCSSContent?: boolean; includeRuntimeId?: boolean } = {}) {
```
Comment (AviVahl) [inline packages/e2e-test-kit/src/browser-functions.ts:12]: next version of typescript (4.2) may fix this for us.

File: tsconfig.json [hunk 3]
```diff
@@ -1,69 +1,40 @@
...
-    // "inlineSourceMap": true,               /* Emit a single file with source maps instead of having a separate file. */
-    // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */
-
-    /* Experimental Options */
-    // "experimentalDecorators": true,        /* Enables experimental support for ES7 decorators. */
-    // "emitDecoratorMetadata": true,         /* Enables experimental support for emitting type metadata for decorators. */
-  }
-}
\ No newline at end of file
+  "extends": "./tsconfig.base.json",
```
Comment (AviVahl) [inline tsconfig.json:2]: this is only required for `ts-tools/node`. will be removed once we run tests from `dist` (future PR)

**Review by barak007 — APPROVED** [review]

**Pull request merged**